File: src/test/java/ru/yandex/praktikum/pages/OrderStatusPage.java
package ru.yandex.praktikum.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class OrderStatusPage {
    private WebDriver driver;
    private WebDriverWait wait;

    private final By orderStatusTitle = By.xpath("//div[contains(text(), 'Статус заказа')]");

    public OrderStatusPage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public boolean isOrderStatusPage() {
        try {
            wait.until(ExpectedConditions.visibilityOfElementLocated(orderStatusTitle));
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: pages/order_status_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from pages.base_page import BasePage

class OrderStatusPage(BasePage):
    ORDER_STATUS_TITLE = (By.XPATH, "//div[contains(text(), 'Статус заказа')]")

    def is_order_status_page(self):
        try:
            self.wait.until(EC.visibility_of_element_located(self.ORDER_STATUS_TITLE))
            return True
        except:
            return False
